bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/AccountController.java

```diff
@@ -0,0 +1,195 @@
+        }
+        if (this.sunAccount.getStatus() == 404) {
+            logger.debug("Sun Synk Account could not be authenticated: 404 {} {}.", this.sunAccount.getError(),
+                    this.sunAccount.getPath());
+            throw new SunSynkAuthenticateException("Sun Synk Accountfailed to authenticate: 404 Not Found.");
+        }
+        getToken();
+    }
+
+    private void getToken() throws SunSynkAuthenticateException {
+        APIdata data = this.sunAccount.getData();
+        APIdata.staticAccessToken = data.getAccessToken();
+    }
+
+    /**
+     * Discovers a list of all inverter tied to a Sunsynk Connect Account
+     * 
+     * @return List of connected inveters
+     * @throws SunSynkInverterDiscoveryException
+     */
+    @SuppressWarnings("unused")
+    public ArrayList<Inverter> getDetails() throws SunSynkInverterDiscoveryException {
+        Details output = new Details();
+        ArrayList<Inverter> inverters = new ArrayList<>();
+        try {
+            Gson gson = new Gson();
+            Properties headers = new Properties();
+            String response = "";
+            String httpsURL = makeLoginURL(
+                    "api/v1/inverters?page=1&limit=10&total=0&status=-1&sn=&plantId=&type=-2&softVer=&hmiVer=&agentCompanyId=-1&gsn=");
+            headers.setProperty("Accept", "application/json");
+            headers.setProperty("Authorization", "Bearer " + APIdata.staticAccessToken);
+            response = HttpUtil.executeUrl(HttpMethod.GET.asString(), httpsURL, headers, null, "application/json",
+                    TIMEOUT_IN_MS);
+            @Nullable
+            Details maybeDeats = gson.fromJson(response, Details.class);
+            if (maybeDeats == null) {
+                throw new SunSynkInverterDiscoveryException("Failed to discover Inverters");
+            }
+            output = maybeDeats;
+        } catch (IOException | JsonSyntaxException e) {
+            if (logger.isDebugEnabled()) {
+                String message = Objects.requireNonNullElse(e.getMessage(), "unkown error message");
+                Throwable cause = e.getCause();
+                String causeMessage = cause != null ? Objects.requireNonNullElse(cause.getMessage(), "unkown cause")
+                        : "unkown cause";
+                logger.debug("Error attempting to find inverters registered to account: Msg = {}. Cause = {}.", message,
+                        causeMessage);
+            }
+            throw new SunSynkInverterDiscoveryException("Failed to discover Inverters", e);
+        }
+        inverters = output.getInverters(APIdata.staticAccessToken);
+        return inverters;
+    }
+
+    private static String makeLoginURL(String path) {
+        return "https://api.sunsynk.net" + "/" + path;
+    }
+
+    private static String makeLoginBody(String username, String password) {
+        return "{\"username\": \"" + username + "\", \"password\": \"" + password
```

Comment:
It would be better to create a DTO class with these fields and serialize it with gson.
Not a blocking issue, a suggestion for maintainability.

Reply:
Add three  new dto:  SunSynkLogin, TokenRefresh and SettingsCommand